src/ThreadedExecutor.h

```diff
@@ -0,0 +1,58 @@
+/* Copyright (C) 2018 European Spallation Source, ERIC. See LICENSE file */
+//===----------------------------------------------------------------------===//
+///
+/// \file
+///
+/// \brief Simple threaded executor to simplify the use of threading in this lib
+/// library.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <concurrentqueue/concurrentqueue.h>
+#include <functional>
+#include <future>
+#include <memory>
+#include <thread>
+
+class ThreadedExecutor {
+private:
+public:
+  using WorkMessage = std::function<void()>;
+  ThreadedExecutor(bool LowPrioThreadExit = false)
```

Comment:
Unless they are in common use, I think using abbreviations only hurts readability. Would prefer `Priority` over `Prio`.

Reply:
Ok.